src/ui/input_handler.rs

```diff
@@ -92,14 +102,73 @@ impl InputHandler {
                 code: KeyCode::Enter,
                 ..
             } => {
-                self.text_insertion = false;
+                self.input_mode = InputMode::Normal;
                 app.on_search();
                 app.on_toggle_popup();
             }
             _ => (),
         }
     }
 
+    fn handle_key_in_keymap_mode<A: Application>(&mut self, key_event: KeyEvent, app: &mut A) {
+        match key_event {
+            KeyEvent {
+                code: KeyCode::Esc, ..
+            }
+            | KeyEvent {
+                code: KeyCode::Enter,
+                ..
+            }
+            | KeyEvent {
+                code: KeyCode::F(1),
+                ..
+            }
+            | KeyEvent {
+                code: KeyCode::Char('?'),
+                ..
+            }
+            | KeyEvent {
+                code: KeyCode::Char('c'),
+                modifiers: KeyModifiers::CONTROL,
+            } => {
+                self.input_mode = InputMode::Normal;
```

Comment:
using Esc or Ctrl+c when popup is visible closes application and I think your intent was to close popup only, am I right?

Reply:
That was the intention, yes. IIRC it seemed to work, when I tested it. Same goes for the new version (where all other keys close the popup).

If there is some edge case, I think it would also affect the search popup, since I largely copied what it does here.

Reply:
it must have been my terminal or something, because the issue disappear 🤷‍♂️ I am going to observe it